tests/unit/parser.js

```diff
@@ -924,6 +924,42 @@ exports.comma = function (test) {
   test.done();
 };
 
+exports["gh-2587"] = function (test) {
+
+  var ops = ["==", "!="];
+  var options = [{}, {eqeqeq: true}, {eqnull: true}, {eqeqeq: true, eqnull: true}];
+
+  ops.forEach(function(op) {
+    options.forEach(function(option) {
```

Comment:
People will be reading this test far more often than they will be trying to modify it. I'd much prefer a simpler style that declare expected input and output without abstraction, even if it means less-than-exhaustive coverage.